miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/analytics/MiniAppAnalytics.kt

```diff
@@ -17,34 +17,86 @@ private inline fun <T> whenHasAnalytics(callback: () -> T) {
 
 /** Only init when analytics dependency is provided. */
 @Suppress("SwallowedException", "TooGenericExceptionCaught")
-internal class MiniAppAnalytics(val rasProjectId: String) {
+internal class MiniAppAnalytics(
+    val rasProjectId: String,
+    private val configs: List<MiniAppAnalyticsConfig>
+) {
 
     companion object {
-        var instance: MiniAppAnalytics? = null
-
-        fun init(rasProjectId: String) = whenHasAnalytics {
-            instance = MiniAppAnalytics(rasProjectId)
+        @Suppress("LongMethod")
+        internal fun sendAnalyticsDefault(
+            rasProjectId: String,
+            eType: Etype,
+            actype: Actype,
+            miniAppInfo: MiniAppInfo?
+        ) = try {
+            val params = createParams(
+                rasProjectId = rasProjectId,
+                acc = BuildConfig.ANALYTICS_ACC,
+                aid = BuildConfig.ANALYTICS_AID,
+                actype = actype,
+                miniAppInfo = miniAppInfo
+            )
+            whenHasAnalytics {
+                RatTracker.event(eType.value, params).track()
+            }
+        } catch (e: Exception) {
+            Log.e("MiniAppAnalytics", e.message.orEmpty())
         }
-    }
 
-    fun sendAnalytics(eType: Etype, actype: Actype, miniAppInfo: MiniAppInfo?) = try {
-        val params = mutableMapOf<String, Any>()
-        // Send to this acc/aid
-        params["acc"] = BuildConfig.ANALYTICS_ACC
-        params["aid"] = BuildConfig.ANALYTICS_AID
+        /** common function to create params to send to tracker. */
+        @Suppress("LongMethod")
+        private fun createParams(
+            rasProjectId: String,
+            acc: Int,
+            aid: Int,
+            actype: Actype,
```

Comment:
We can rename it to `acType` to preserve camelCase style.

Reply:
i think its better to be like this because there is  a predefined parameter for `RatTracker` actype, so it can avoid confusion, and also this variable is not defined by me, it was already there @khairul-alam-licon

Reply:
Oh I see.